src/data/move-attrs/stat-stage-change-all-opps.ts

```diff
@@ -0,0 +1,130 @@
+import { MoveCategory } from "#enums/move-category";
+import { type BattleStat, Stat } from "#enums/stat";
+import { PlayerPokemon, type Pokemon } from "#app/field/pokemon";
+import { globalScene } from "#app/global-scene";
+import { StatStageChangePhase } from "#app/phases/stat-stage-change-phase";
+import { AttackMove } from "../move";
+import type { Move } from "#app/data/move";
+import { type MoveEffectAttrOptions, MoveEffectAttr } from "#app/data/move-attrs/move-effect-attr";
+import type { MoveConditionFunc } from "../move-conditions";
+
+/**
+ * Set of optional parameters that may be applied to stat stage changing effects
+ * @extends MoveEffectAttrOptions
+ * @see {@linkcode StatStageChangeAttr}
+ */
+interface StatStageChangeAllOppsAttrOptions extends MoveEffectAttrOptions {
+  /** If defined, needs to be met in order for the stat change to apply */
+  condition?: MoveConditionFunc;
+  /** `true` to display a message */
+  showMessage?: boolean;
+}
+
+/**
+ * Attribute used for G-Max moves that change the stats of all opponents
+ *
+ * +-------------+----------+--------+
+ * | G-Max Move  |   Stat   | Change |
+ * +-------------+----------+--------+
+ * | Foam Burst  | Speed    |     -2 |
+ * | Tartness    | Evasion  |     -1 |
+ * +-------------+----------+--------+
+ *
+ * @param stats {@linkcode BattleStat} Array of stat(s) to change
+ * @param stages How many stages to change the stat(s) by, [-6, 6]
+ * @param selfTarget `true` if the move is self-targetting
+ * @param options {@linkcode StatStageChangeAttrOptions} Container for any optional parameters for this attribute.
+ *
+ * @extends MoveEffectAttr
+ */
+
+export class StatStageChangeAllOppsAttr extends MoveEffectAttr {
+  public stats: BattleStat[];
+  public stages: number;
+  /**
+   * Container for optional parameters to this attribute.
+   * @see {@linkcode StatStageChangeAttrOptions} for available optional params
+   */
+  protected override options?: StatStageChangeAllOppsAttrOptions;
+
+  constructor(stats: BattleStat[], stages: number, selfTarget?: boolean, options?: StatStageChangeAllOppsAttrOptions) {
+    super(selfTarget, options);
+    this.stats = stats;
+    this.stages = stages;
+    this.options = options;
+  }
```

Comment:
```suggestion
  constructor(stats: BattleStat[], stages: number, options?: StatStageChangeAllOppsAttrOptions) {
    super(true, options);
    this.stats = stats;
    this.stages = stages;
    this.options = options;
  }
```

This should always be self-targeted because:
1. The stat drop should trigger even if the target faints (affecting the target's ally)
2. The stat drop should affect opponents that are behind substitutes